### ISSUE TITLE
[bug] false positive of rule ensure-use-memo-has-non-empty-deps

### ISSUE THREAD

**Issue by Jack-Works** [Rel1cx/eslint-react#530]
### Describe the bug

```js

        const [width, setWidth] = useState<undefined | number>(undefined)
        const [open, setOpen] = useState<boolean>(false)
        const [title, setTitle] = useState<string | undefined>(undefined)

        const refItem = useMemo(() => {
            return {
                setWidth,
                setWrap: setOpen,
                setWrapperName: setTitle,
            }
        }, [])
```

this should not trigger the lint error because it cannot be lifted out of the component

### Platform and versions

```Text
"@eslint-react/eslint-plugin": "^1.5.13-next.1"
```

**Comment by SukkaW** [comment]
`eslint-plugin-react-hooks` automatically excludes `setWidth`, `setOpen`, `setTitle` out of the dep array, hence the issue.

But that leads to a question: is the `useMemo` really needed in the first place?

**Comment by Jack-Works** [comment]
> `eslint-plugin-react-hooks` automatically excludes `setWidth`, `setOpen`, `setTitle` out of the dep array, hence the issue.
> 
> But that leads to a question: is the `useMemo` really needed in the first place?

Yes. In this case, `refItem` is used as the ref of this component (like HTMLInputElement exposes) it should be stable if possible but cannot be lifted.
